### PR TITLE
Move validation inside getOverview...

### PR DESCRIPTION
**Pull request by labrys**
and clean up TV.py a little bit.
- Moved validation inside `getOverview` to simplify usage.
- Added a debug entry if `getOverview` fails to get a valid overview
- Streamlined `TVShow` properties: `is_anime`, `is_sports`, `is_scene`
- Refactored any and best qualities in common.py to allowed and preferred for consistency

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked
- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

File: sickbeard/webapi.py [hunk 1]
```diff
@@ -1159,7 +1159,7 @@ def run(self):
 
             for curResult in sql_results:
 
-                cur_ep_cat = curShow.getOverview(int(curResult["status"] or -1))
+                cur_ep_cat = curShow.getOverview(curResult["status"])
```
Comment (fernandog) [inline sickbeard/webapi.py]: If status is -1,   INT will fail. We had tons of that issue in past

File: sickbeard/tv.py [hunk 2]
```diff
@@ -149,24 +149,15 @@ def __init__(self, indexer, indexerid, lang=""):
 
     @property
     def is_anime(self):
-        if int(self.anime) > 0:
-            return True
-        else:
-            return False
+        return True if int(self.anime) > 0 else False
```
Comment (MGaetan89) [inline sickbeard/tv.py:152]: `return int(self.anime) > 0`? Same for the two below.

File: sickbeard/tv.py [hunk 3]
```diff
@@ -1277,26 +1268,34 @@ def wantEpisode(self, season, episode, quality, manualSearch=False, downCurQuali
...
     def getOverview(self, epStatus):
+        """
+        Get the Overview status from the Episode status
+
+        :param epStatus: an Episode status
+        :return: an Overview status
+        """
 
-        if epStatus == WANTED:
+        ep_status = try_int(epStatus) or UNKNOWN
```
Comment (MGaetan89) [inline sickbeard/tv.py:1278]: `try_int(epStatus, UNKNOWN)`?

**Comment by fernandog** [comment]
Unknown = -1 afaik

File: sickbeard/tv.py (hunk 3 above)
Comment (labrys) [inline sickbeard/tv.py:1278]: No because then it could return a 0 which we don't want

**Comment by labrys** [comment]
Correct UNKNOWN is -1

File: sickbeard/tv.py (hunk 2 above)
Comment (labrys) [inline sickbeard/tv.py:152]: Was going to do that but I was rushing out the door, this was more benign and didn't require thought :).  Obviously they are equivalent tho and should be changed to the more concise form

**Pull request merged**